File: assignments/assign01/assn01_ws/src/heatControl_py/heatControl_py/tempSensor.py
#!/usr/bin/env python3
import rclpy   # rclpy must be declared as a dependence in package.xml - this should be added automatically when you create the package
from rclpy.node import Node

from example_interfaces.msg import Int32
from rcl_interfaces.msg import ParameterDescriptor
import random

class TempSensor(Node):
    def __init__(self):
        super().__init__("tempSensor")

        # declare parameter
        period_parameter_descriptor = ParameterDescriptor(description='timer period value')
        self.declare_parameter('period', 1, period_parameter_descriptor)

        # create publisher
        self.tempSensorPublisher_ = self.create_publisher(Int32, "currentTemp", 10)

        # create timer
        self.timer_ = self.create_timer(
            self.get_parameter('period').value,
            self.callback_handleTimer
        )

        self.initialize()

    def callback_handleTimer(self):
        self.temperature_: int = random.randint(80,120)
        msg = Int32()
        msg.data = self.temperature_
        self.tempSensorPublisher_.publish(msg)
        self.get_logger().info("current temperature is " + str(self.temperature_))

    def initialize(self):
        self.temperature_ = 0

def main(args = None):
    rclpy.init(args=args)
    node = TempSensor()
    rclpy.spin(node)
    rclpy.shutdown()        
        
if __name__ == "__main__":
    main()  

File: assignments/assign01/assn01_ws/src/heatControl_cpp/src/tempSensor.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int32.hpp"
#include <random>

class TempSensor: public rclcpp::Node
{
public:
   TempSensor() : Node("tempSensor"), temperature_(3)
   {
         tempSensorPublisher_ = this->create_publisher<example_interfaces::msg::Int32>("currentTemp", 10);
         this->declare_parameter("period", 1);
         timer_ = this->create_wall_timer(
           std::chrono::seconds(this->get_parameter("period").as_int()), 
           std::bind(&TempSensor::callback_handleTimer, this)
         );
   }

   void callback_handleTimer()
   {
         std::random_device rd;
         std::mt19937 gen(rd()); // Mersenne Twister engine
         std::uniform_int_distribution<> dist(80, 120); // range: 1 to 100

         temperature_ = dist(gen);
         auto msg = example_interfaces::msg::Int32();
         msg.data = temperature_;
         tempSensorPublisher_->publish(msg);

         RCLCPP_INFO(this->get_logger(), "the current temperature is %d\n", temperature_);
   }
private:
   int temperature_;
   rclcpp::Publisher<example_interfaces::msg::Int32>::SharedPtr tempSensorPublisher_;
   rclcpp::TimerBase::SharedPtr timer_;
};


int main(int argc, char **argv)
{
   rclcpp::init(argc, argv);
   auto node = std::make_shared<TempSensor>();
   rclcpp::spin(node);
   rclcpp::shutdown();
   return 0;
}